Sources/Sentry/SentryUIApplication.m

```diff
@@ -4,7 +4,34 @@
 
 #    import <UIKit/UIKit.h>
 
-@implementation SentryUIApplication
+@implementation SentryUIApplication {
+    UIApplicationState appState;
+}
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        [NSNotificationCenter.defaultCenter addObserver:self
```

Comment:
`m`: Please use the SentryNSNotificationCenterWrapper and add some tests when this solves the problem.